catalog/spatial/registry/registry-federation-admin-impl/src/main/java/org/codice/ddf/registry/federationadmin/impl/FederationAdmin.java

```diff
@@ -0,0 +1,699 @@
+            LOGGER.warn("Couldn't get remote registry metacards '{}'", e);
+        }
+
+        if (customSlots != null) {
+            nodes.put(CUSTOM_SLOTS_KEY, customSlots);
+        }
+
+        Map<String, Object> autoPopulateMap = new HashMap<>();
+        autoPopulateMap.put(SERVICE_BINDINGS_KEY, endpointMap.values());
+        nodes.put(AUTO_POPULATE_VALUES_KEY, autoPopulateMap);
+
+        nodes.put(NODES_KEY, registryMetacardInfo);
+
+        return nodes;
+    }
+
+    private List<Map<String, Object>> getWebMapsFromRegistryPackages(
+            List<RegistryPackageType> packages, Map<String, Metacard> metacardByRegistryIdMap) {
+        List<Map<String, Object>> registryMaps = new ArrayList<>();
+        for (RegistryPackageType registryPackage : packages) {
+            Map<String, Object> registryWebMap = registryMapConverter.convert(registryPackage);
+
+            Metacard metacard = metacardByRegistryIdMap.get(registryPackage.getId());
+            Map<String, Object> transientValues = getTransientValuesMap(metacard);
+            if (MapUtils.isNotEmpty(transientValues)) {
+                registryWebMap.put(TRANSIENT_VALUES_KEY, transientValues);
+            }
+
+            if (MapUtils.isNotEmpty(registryWebMap)) {
+                registryMaps.add(registryWebMap);
+            }
+        }
+        return registryMaps;
+    }
+
+    private Metacard getRegistryMetacardFromRegistryPackage(RegistryPackageType registryPackage)
+            throws FederationAdminException {
+        if (registryPackage == null) {
+            throw new FederationAdminException(
+                    "Error creating metacard from registry package. Null package was received.");
+        }
+        Metacard metacard;
+
+        try {
+            JAXBElement<RegistryPackageType> jaxbRegistryObjectType =
+                    EbrimConstants.RIM_FACTORY.createRegistryPackage(registryPackage);
+
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            parser.marshal(marshalConfigurator, jaxbRegistryObjectType, baos);
+            InputStream xmlInputStream = new ByteArrayInputStream(baos.toByteArray());
+            metacard = registryTransformer.transform(xmlInputStream);
+
+        } catch (IOException | CatalogTransformerException | ParserException e) {
+            String message = "Error creating metacard from registry package.";
+            LOGGER.error("{} Registry id: {}", message, registryPackage.getId());
+            throw new FederationAdminException(message, e);
+        }
+
+        return metacard;
+    }
+
+    private Metacard getRegistryMetacardFromInputStream(InputStream inputStream)
+            throws FederationAdminException {
+        if (inputStream == null) {
+            throw new FederationAdminException(
+                    "Error converting input stream to a metacard. Null input stream provided.");
+        }
+
+        Metacard metacard;
+        try {
```

Comment:
Since they are ByteArrayInput/Output stream they don't need to be closed. I will add a IOUtils.closeQuietly to them though so static analysis tools don't complain.